OrbitBase/include/OrbitBase/UniqueResource.h

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_UNIQUE_RESOURCE_H_
+#define ORBIT_BASE_UNIQUE_RESOURCE_H_
+
+#include <functional>
+#include <optional>
+
+namespace OrbitBase {
+
+/* unique_resource helps to manage a unique identity which is not a pointer,
+   so unique_ptr cannot be used. Typical examples are file or window handles
+   from C libraries. The resource (handle) is typically small and trivially
+   copyable, but that's not required. Though we require it to be noexcept
+   moveable.
+
+   Unlike unique_ptr this resource requires you to provide a Deleter type.
+   No default deleter exists.
+
+   Empty base class optimization for Deleter is not implemented (yet).
+*/
+template <typename Handle, typename Deleter_>
+class unique_resource {
+ public:
+  using Resource = Handle;
+  using Deleter = Deleter_;
+
+  constexpr unique_resource() {}
+  constexpr unique_resource(Handle handle, Deleter deleter = Deleter{})
```

Comment:
Shouldn't it be `Resource` here, if you define the type alias above?
Otherwise, I'd expect `Deleter_`

Reply:
Semantically it makes no difference, but for the sake of consistency we should use Resource_/Resource and Deleter_/Deleter. I will fix that. Thanks.